core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClientBuilder.java

```diff
@@ -38,43 +37,53 @@
  */
 public abstract class AbstractRetryingClientBuilder<O extends Response> {
 
-    @Nullable
-    private final RetryRule retryRule;
+    @Nullable private final RetryConfigBuilder<O> retryConfig;
 
-    @Nullable
-    private final RetryRuleWithContent<O> retryRuleWithContent;
-
-    private int maxTotalAttempts = Flags.defaultMaxTotalAttempts();
-    private long responseTimeoutMillisForEachAttempt = Flags.defaultResponseTimeoutMillis();
+    private RetryConfigMapping<O> mapping;
 
     /**
      * Creates a new builder with the specified {@link RetryRule}.
      */
     AbstractRetryingClientBuilder(RetryRule retryRule) {
-        this(requireNonNull(retryRule, "retryRule"), null);
+        retryConfig = RetryConfig.builder(requireNonNull(retryRule, "retryRule"));
+        mapping = (ctx, req) -> retryConfig.build();
```

Comment:
We should not call `.build()` on each request. Could we build it only once?

Reply:
I made mapping nullable and changed its getter to call retryConfig() if it's null, so that we keep the assumption that mapping() always returns a mapping (not null). If a fixed config was set, the mapping always returns this config.